### ISSUE TITLE
Make UUIDs unpredictable

### ISSUE THREAD

**Issue by mozfreddyb** [mozilla-services/screenshots#2299]
Pageshot's current UUID implementations [[1]](https://github.com/mozilla-services/pageshot/blob/a2d5a738dd3c2ce8a45d795f878efac7c0784d7f/webextension/makeUuid.js#L1) [[2]](https://github.com/mozilla-services/pageshot/blob/a2d5a738dd3c2ce8a45d795f878efac7c0784d7f/shared/shot.js#L162) use `Math.random` to generate UUIDs.
This method produces predictable IDs and someone could compute your next UUIDs based on previous. This might allow an attacker to crawl all shots.

I'm not sure how bad this is for your threat model (probably not *so* bad?), but I wanted to make sure this is noted.

**Comment by ianb** [comment]
I actually had your notes from this somewhere but can't find them now, and forgot to make a ticket.  Since the unguessability of these strings is important we should make this change.

We should also change randomString.js, and do a search for Math.random in case there's another instance I've forgotten.

**Comment by 6a68** [comment]
`crypto.getRandomValues()` should work in a worker or a regular DOM page: https://developer.mozilla.org/en-US/docs/Web/API/RandomSource/getRandomValues

**Comment by mozfreddyb** [comment]
Ah, there was an IRC conversation. I didn't figure this was related.

Here's something along the lines of what I wrote to you
```js
function hexString(length) {
  return Array.from(  // wrap this into a real array
    crypto.getRandomValues(new Uint8Array(length/2)) // random uint8 array
  ).map(
    (el) => el.toString(16).padStart(2, "0") // hex, those <127 padded with
leading 0.
  ).join("")
}
```

**Comment by ianb** [comment]
Ah, I threw it in a comment on a not-very-related bug: https://github.com/mozilla-services/pageshot/issues/2139#issuecomment-277037177

**Comment by 6a68** [comment]
Closed by #2404 (thanks, @garbados!)

**Comment by garbados** [comment]
Heck, glad to help :)
